### PR TITLE
feat: macOS wireless plugin

### PR DESCRIPTION
**Pull request by davidgs**
Adds support for monitoring wireless networks on macOS

### CODE REVIEW

**Comment by powersj** [comment]
@davidgs,

Can you do two things:

1. Run `make fmt` and push again. There is an unused import.
2. Rebase on master, so your tests run on Windows. I put in a fix yesterday.

Thanks!

**Comment by davidgs** [comment]
@powersj do the tests need to run on windows? The plug-in won't ever run there (since it's macOS only)

**Comment by Hipska** [comment]
> I understand it would be nicer but it is much harder to do. I can look at it, but it will take longer.

No problem, take your time. 

Ps, next feature release is planned somewhere in March.

File: plugins/inputs/wireless/wireless_mac.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+func (w *Wireless) Gather(acc telegraf.Accumulator) error {
+	if runtime.GOOS == "darwin" {
+		// collect MAC OS wireless data
+		wg := new(sync.WaitGroup)
+		wg.Add(3)
+		wireless, err := w.exe_cmd(OSXCMD, wg)
+		if err != nil {
+			return err
+		}
+		metrics, tags, err := w.loadMacWirelessTable(wireless, w.DumpZeros)
```
Comment (Hipska) [inline plugins/inputs/wireless/wireless_mac.go:40]: No need to pass `DumpZeros` as it is available inside the method as well..

File: plugins/inputs/wireless/wireless_mac.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+			return err
+		}
+		acc.AddGauge("wireless", metrics, tags)
+		return nil
+	}
+	return errors.New("OS Not Supported")
+}
+
+func (w *Wireless) loadMacWirelessTable(table []byte, dumpZeros bool) (map[string]interface{}, map[string]string, error) {
+	metrics := strings.Split(string(string(table)), "\n")
```
Comment (Hipska) [inline plugins/inputs/wireless/wireless_mac.go:51]: Why is this double casting to string needed?

File: plugins/inputs/wireless/wireless_mac.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	out, err := exec.Command(head, parts...).Output()
+	wg.Done() // Need to signal to waitgroup that this goroutine is done
+	return out, err
+}
+
+func (w *Wireless) Gather(acc telegraf.Accumulator) error {
+	if runtime.GOOS == "darwin" {
+		// collect MAC OS wireless data
+		wg := new(sync.WaitGroup)
+		wg.Add(3)
```
Comment (Hipska) [inline plugins/inputs/wireless/wireless_mac.go:35]: Is this wait group actually needed? What does it add?

File: plugins/inputs/wireless/wireless_mac.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	}
+	tags["interface"] = "airport"
+	return points, tags, nil
+}
+
+func init() {
+	inputs.Add("wireless_mac", func() telegraf.Input {
+		return &Wireless{}
+	})
+}
```
Comment (Hipska) [inline plugins/inputs/wireless/wireless_mac.go:78]: init already exists in `wireless.go`

File: plugins/inputs/wireless/wireless_notlinux.go [hunk 5]
```diff
@@ -1,5 +1,5 @@
-//go:build !linux
-// +build !linux
+//go:build !linux && !darwin
+// +build !linux,!darwin
```
Comment (Hipska) [inline plugins/inputs/wireless/wireless_notlinux.go:2]: Maybe rename this file to `wireless_notsupported.go`?

**Review by Hipska — CHANGES_REQUESTED** [review]
I see a lot of unrelated changes in this PR, these shouldn't exist in here. They also make reviewing a lot harder.

Please mark this PR as draft if you are not finished and do not expect reviews yet.

**Comment by davidgs** [comment]
I'm closing this garbage PR and will resubmit.

**Comment by Hipska** [comment]
Well, actually it already looked better after this last merge..

**Pull request closed without merging**